src/util/styles/style-object.js

```diff
@@ -303,15 +303,42 @@ class StyleObject {
 		const hasHover = hover === 'all' || ( Array.isArray( hover ) && hover.includes( 'hover' ) )
 		const hasParentHover = hover === 'all' || ( Array.isArray( hover ) && hover.includes( 'parent-hover' ) )
 
-		let parentHoverSelector = `:where(.stk-block:hover) .%s ${ selector }`
-		let hoverSelector = _hoverSelector || `.stk-block.%s:hover ${ selector }`
+		let parentHoverSelector
+		let hoverSelector
 
-		// This is for the editor, change the selector to make the styles show up right away.
-		if ( blockState === 'hover' ) {
-			hoverSelector = `.%s.stk--is-hovered ${ selector }`
+		if ( Array.isArray( _hoverSelector ) && blockState !== 'hover' ) {
+			hoverSelector = _hoverSelector.join( ', ' )
 		}
-		if ( blockState === 'parent-hovered' ) {
-			parentHoverSelector = `.%s.stk--is-hovered .%s-container`
+
+		if ( Array.isArray( selector ) ) {
+			// This is to handle multiple selectors.
+			parentHoverSelector = selector.map( s => {
+				if ( blockState === 'parent-hovered' ) {
+					return `.%s.stk--is-hovered .%s-container`
+				}
+
+				return `:where(.stk-hover-parent:hover) .%s ${ s }`
+			} ).join( ', ' )
+
+			hoverSelector = hoverSelector || selector.map( s => {
+				if ( blockState === 'hover' ) {
+					return `.%s.stk--is-hovered ${ s }`
+				}
+
+				return `.stk-block.%s:hover ${ s }`
+			} ).join( ', ' )
+
+			selector = selector.join( ', ' )
+		} else {
+			parentHoverSelector = `:where(.stk-hover-parent:hover) .%s ${ selector }`
+			hoverSelector = _hoverSelector || `.stk-block.%s:hover ${ selector }`
+			// This is for the editor, change the selector to make the styles show up right away.
+			if ( blockState === 'hover' ) {
+				hoverSelector = `.%s.stk--is-hovered ${ selector }`
+			}
+			if ( blockState === 'parent-hovered' ) {
+				parentHoverSelector = `.%s.stk--is-hovered ${ selector }`
+			}
```

Comment:
This looks a bit messy. Can you make a function with the structure:

```js
const parependClass = ( prependString, selector ) => {
    let combinedSelector = ''
    // ...Handle if selector is a string or array
    return combinedSelector
}
```

The function can do the splitting and appending of the class. It should already handle whether the given selector is a string or an array.

It can be called like this:
```js
parentHoverSelector = parependClass( ':where(.stk-hover-parent:hover) .%s', selector )
hoverSelector = _hoverSelector || parependClass( '.stk-block.%s:hover', selector )
```